### ISSUE TITLE
API: HTTP 500 when creating an album with duplicate asset_ids

### ISSUE THREAD

**Issue by seasox** [immich-app/immich#8439]
### The bug

When POSTing an album using the `/api/album` endpoint with a duplicate UUID in assetIds, an internal server error due to unique key constraint violation in `album_assets_asset`.

I am currently bulk-uploading a lot of images into immich. There are some duplicates in my files, which are correctly identified while uploading assets using the `POST /api/asset` endpoint with the `duplicate` flag set to true and the asset's UUID returned. I collect the UUIDs from the `POST` calls and add them to a number of albums afterwards. Sometimes, there is a duplicate in the assetIds parameter in the `POST /api/album`, which results in an Internal Server Error. The client-side fix for this is to make sure that assetIds does not contain duplicates, but I think it would be better to either handle duplicate asset UUIDs gracefully or improve the error message.

### The OS that Immich Server is running on

Docker on macOS

### Version of Immich Server

v1.100.0

### Version of Immich Mobile App

v1.100.0

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
#
# WARNING: Make sure to use the docker-compose.yml of the current release:
#
# https://github.com/immich-app/immich/releases/latest/download/docker-compose.yml
#
# The compose file on main may not be compatible with the latest release.
#

name: immich

services:
  immich-server:
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    command: ['start.sh', 'immich']
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
    env_file:
      - .env
    ports:
      - 2283:3001
    depends_on:
      - redis
      - database
    restart: always

  immich-microservices:
    container_name: immich_microservices
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    # extends: # uncomment this section for hardware acceleration - see https://immich.app/docs/features/hardware-transcoding
    #   file: hwaccel.transcoding.yml
    #   service: cpu # set to one of [nvenc, quicksync, rkmpp, vaapi, vaapi-wsl] for accelerated transcoding
    command: ['start.sh', 'microservices']
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
    env_file:
      - .env
    depends_on:
      - redis
      - database
    restart: always

  immich-machine-learning:
    container_name: immich_machine_learning
    # For hardware acceleration, add one of -[armnn, cuda, openvino] to the image tag.
    # Example tag: ${IMMICH_VERSION:-release}-cuda
    image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}
    # extends: # uncomment this section for hardware acceleration - see https://immich.app/docs/features/ml-hardware-acceleration
    #   file: hwaccel.ml.yml
    #   service: cpu # set to one of [armnn, cuda, openvino, openvino-wsl] for accelerated inference - use the `-wsl` version for WSL2 where applicable
    volumes:
      - model-cache:/cache
    env_file:
      - .env
    restart: always

  redis:
    container_name: immich_redis
    image: registry.hub.docker.com/library/redis:6.2-alpine@sha256:51d6c56749a4243096327e3fb964a48ed92254357108449cb6e23999c37773c5
    restart: always

  database:
    container_name: immich_postgres
    image: registry.hub.docker.com/tensorchord/pgvecto-rs:pg14-v0.2.0@sha256:90724186f0a3517cf6914295b5ab410db9ce23190a2d9d0b9dd6463e3fa298f0
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
    volumes:
      - pgdata:/var/lib/postgresql/data
    ports:
      - 5432:5432
    restart: always

volumes:
  pgdata:
  model-cache:
```

### Your .env content

```Shell
# You can find documentation for all the supported env variables at https://immich.app/docs/install/environment-variables

# The location where your uploaded files are stored
UPLOAD_LOCATION=./library

# The Immich version to use. You can pin this to a specific version like "v1.71.0"
IMMICH_VERSION=release

# Connection secret for postgres. You should change it to a random password
DB_PASSWORD=postgres

# The values below this line do not need to be changed
###################################################################################
DB_HOSTNAME=immich_postgres
DB_USERNAME=postgres
DB_DATABASE_NAME=immich

REDIS_HOSTNAME=immich_redis
```

### Reproduction steps

```bash
1. Create an image using the /api/asset endpoint. Let `A` be the UUID generated by the immich backend.
2. Create an album using the /api/album endpoint with assetIds = [ A, A ], i.e., a duplicate UUID in assetIds.
3. Observe that an 500 Internal Server Error is triggered.
```

**Comment by danieldietzler** [comment]
1) you aren't supposed to manually set a UUID. That should always be generated automatically.
2) What's even the issue?

**Comment by seasox** [comment]
1. I am currently pushing a lot of images using the API. Some of them are duplicates, i.e., subsequent `POST /api/asset` calls result in responses with the `duplicate` flag set and return the previously inserted UUID with the `duplicate` flag set.
2. I spent quite some time debugging this issue while automatically adding images to albums. I think it would be good to either handle the duplicate UUIDs gracefully or improve the error message provided.

**Comment by seasox** [comment]
I think this is a misunderstanding. I'm not setting the UUID field myself, I collect the auto-generated ones from the responses to my `POST /api/asset` calls. As far as I can see, there isn't even a way to manually set the UUID in the `POST /api/asset` call (which makes sense).

I have updated the issue to hopefully better describe the issue.

**Comment by jrasm91** [comment]
I think this has already been fixed on main.

**Comment by jrasm91** [comment]
Ah this is specifically for create album. This is still an issue, yes.

**Comment by jrasm91** [comment]
Fixed in #8686.
